File: 프로그래머스/unrated/155651. 호텔 대실/호텔 대실.java
class Solution {
    public int solution(String[][] book_time) {
        int answer = 0;
        int arr[] = new int[1440];

        for(String[] time : book_time){
            int start = change(time[0]);
            int end = Math.min(1439, change(time[1]) + 10);
            for(int i = start; i < end; i++){
                arr[i]++;
            }
        }

        for(int count : arr){
            answer = Math.max(count, answer);
        }
        
        return answer;
    }
    
    public int change(String time){
        String[] timeArr = time.split(":");
        return Integer.parseInt(timeArr[0]) * 60 + Integer.parseInt(timeArr[1]); 
    }
}

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def solution(book_time):
    answer = 0
    max_minute = [0] * 1440
    
    for time in book_time:
        start = change(time[0])
        end = min(1439,change(time[1]) + 10)
        
        for i in range(start,end):
            max_minute[i] += 1
            
    for j in max_minute:
        answer = max(answer,j)
    
    return answer

def change(time):
    arr = time.split(":")
    #분으로 변환해서 반환
    return int(arr[0]) * 60 + int(arr[1])
        